client/my-sites/stats/stats-purchase/stats-pwyw-uprade-slider/index.tsx

```diff
@@ -112,30 +94,35 @@ function StatsPWYWUpgradeSlider( {
 	const defaultAveragePayment = defaultStartingValue * settings.sliderStepPrice;
 	const uiStrings = useTranslatedStrings( defaultAveragePayment, currencyCode );
 
-	let steps = getPWYWPlanTiers( 0, 50 );
-	if ( settings !== undefined ) {
-		steps = stepsFromSettings( settings, currencyCode || '' );
-	}
+	// New steps generation.
+	const tiers = generatePlanTiers( settings );
+	const steps = generateSteps( tiers, currencyCode, settings );
 	const marks = [ 0, steps.length - 1 ];
 
-	const handleSliderChanged = ( index: number ) => {
+	// New mapped indexing for slider.
+	// Implemented this way so as to not break parent logic.
+	const disableFreeProduct = settings.minSliderPrice !== 0;
+	const mappedDefaultIndex = disableFreeProduct ? defaultStartingValue - 1 : defaultStartingValue;
+
+	// New slider change handler.
+	const handleSliderChanged2 = ( index: number ) => {
```

Comment:
Is the postfix number of the naming existing for any reason?

Reply:
Thanks for flagging that. I was meaning to rename it!